core/jvm/src/main/scala/zio/internal/PlatformSpecific.scala

```diff
@@ -38,6 +40,19 @@ private[internal] trait PlatformSpecific {
       }
     }
 
+  /**
+   * Adds a signal handler that executes the specified action on receiving a signal.
+   */
+  def addSignalHandler(signal: String)(handler: () => Unit): Boolean =
+    try {
+      val _ = Signal.handle(new Signal(signal), _ => handler())
+      true
+    } catch {
+      case e: IllegalArgumentException =>
+        println(s"Failed to install signal: $e")
```

Comment:
I think we don't want to print this and instead the caller can decide how to handle the failure to install the signal.